### PR TITLE
Bugfix [v97] Warnings xcode

### PR DESCRIPTION
**Pull request by lmarceau**
Thanks to @muhasturk for this contribution to fix some warnings in Xcode. Warnings were fixed in [PR 9203](https://github.com/mozilla-mobile/firefox-ios/pull/9203). I addressed a minor comment about a file duplication and rebased on main on my remote.

### CODE REVIEW

File: Client.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -391,6 +390,10 @@
 		8A11C8132731E54800AC7318 /* DictionaryExtensionsTests.swift in Sources */ = {isa = PBXBuildFile; fileRef = 8A11C8122731E54800AC7318 /* DictionaryExtensionsTests.swift */; };
 		8A2783F1275FFDC50080D29D /* KeyboardPressesHandler.swift in Sources */ = {isa = PBXBuildFile; fileRef = 8A2783F0275FFDC50080D29D /* KeyboardPressesHandler.swift */; };
 		8A2825352760399B00395E66 /* KeyboardPressesHandlerTests.swift in Sources */ = {isa = PBXBuildFile; fileRef = 8A2825342760399B00395E66 /* KeyboardPressesHandlerTests.swift */; };
+		8A84D9E1278DED3A0019113B /* PanelDataObservers.swift in Sources */ = {isa = PBXBuildFile; fileRef = 8A84D9E0278DED3A0019113B /* PanelDataObservers.swift */; };
+		8A84D9E2278DED3A0019113B /* PanelDataObservers.swift in Sources */ = {isa = PBXBuildFile; fileRef = 8A84D9E0278DED3A0019113B /* PanelDataObservers.swift */; };
+		8A84D9E3278DED3A0019113B /* PanelDataObservers.swift in Sources */ = {isa = PBXBuildFile; fileRef = 8A84D9E0278DED3A0019113B /* PanelDataObservers.swift */; };
+		8A84D9E4278DED3A0019113B /* PanelDataObservers.swift in Sources */ = {isa = PBXBuildFile; fileRef = 8A84D9E0278DED3A0019113B /* PanelDataObservers.swift */; };
```
Comment (nishant2718) [inline Client.xcodeproj/project.pbxproj:396]: Are these changes just a reorg?

File: WidgetKit/Info.plist [hunk 2]
```diff
@@ -17,7 +17,7 @@
 	<key>CFBundlePackageType</key>
 	<string>$(PRODUCT_BUNDLE_PACKAGE_TYPE)</string>
 	<key>CFBundleShortVersionString</key>
-	<string>1.0</string>
+	<string>0.0.1</string>
```
Comment (nishant2718) [inline WidgetKit/Info.plist:20]: Why is this and the other pList change needed?

Comment (lmarceau) [reply WidgetKit/Info.plist:20]: That's explained by the previous PR comments. To fix the following issue:
Screen Shot 2022-01-11 at 1 41 21 PM

the CFBundleShortVersionString of an app extension ('1.0') must match that of its containing parent app ('0.0.1').

File: Client.xcodeproj/project.pbxproj (hunk 1 above)
Comment (lmarceau) [reply Client.xcodeproj/project.pbxproj:396]: I moved the file in this case! It was referenced in Client and in Providers. With some duplication in code as well 🤔

File: WidgetKit/Info.plist (hunk 2 above)
Comment (nishant2718) [reply WidgetKit/Info.plist:20]: ahh I see. Okay great!

**Review by nishant2718 — APPROVED** [review]
Looks good! As long as BR passes, we should be fine here.

**Comment by lmarceau** [comment]
Reverted the changes about the PanelDataObservers since it messed it up the build in unexpected ways. Let's move along for now and improve on this part later.

**Comment by lmarceau** [comment]
testTopSitesOpenInNewPrivateTab passes locally
Screen Shot 2022-01-11 at 4 35 07 PM

testBookmarksAwesomeBar passes locally
Screen Shot 2022-01-11 at 4 38 31 PM

**Pull request merged**